src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/DocumentProcessor.java

```diff
@@ -0,0 +1,104 @@
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.function.BiConsumer;
+import java.util.stream.StreamSupport;
+import java.util.zip.ZipFile;
+
+/**
+ * Main component that is run in intervals.
+ *
+ * Looks up all available containers for zip files and extracts pdf files from them.
+ */
+@Component
+public class DocumentProcessor {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DocumentProcessor.class);
+
+    private final CloudBlobClient cloudBlobClient;
+    private final BiConsumer<String, ZipFile> handleZip;
+
+    /**
+     * Create new document processor for zip files stored on a blob storage.
+     *
+     * @param cloudBlobClient connection to Azure Blob Storage
+     * @param handleZip callback function invoked for every downloaded zip file.
+     *          See {@link ZipFileHandler#accept(String, ZipFile)}.
+     */
+    @Autowired
+    public DocumentProcessor(CloudBlobClient cloudBlobClient, BiConsumer<String, ZipFile> handleZip) {
+        this.cloudBlobClient = cloudBlobClient;
+        this.handleZip = handleZip;
+    }
+
+    /**
+     * Main method for reading blob files from all containers.
+     */
+    @Scheduled(fixedDelayString = "${scan.delay}")
+    public void readBlobs() {
+        cloudBlobClient.listContainers().forEach(cloudBlobContainer -> {
+            try {
+                String containerName = cloudBlobContainer.getName();
+                LOGGER.info("Processing {} container", containerName);
+                CloudBlobContainer container = cloudBlobClient.getContainerReference(containerName);
+                readBlobsFromContainer(container);
+            } catch (URISyntaxException e) {
+                LOGGER.error("Invalid URL", e);
+                // TODO: track exception in AppInsights Telemetry
+            } catch (StorageException e) {
+                LOGGER.error("Could not obtain container reference", e);
+                // TODO: track exception in AppInsights Telemetry
+            }
+        });
+    }
+
+    private void readBlobsFromContainer(CloudBlobContainer container) {
+        StreamSupport.stream(container.listBlobs().spliterator(), false)
+
+            // checking if it is safe to cast to CloudBlob
+            .filter(CloudBlob.class::isInstance)
+            .map(CloudBlob.class::cast)
+
+            // so that we could get method for retrieving its name
+            .filter(cloudBlob -> cloudBlob.getName().toLowerCase().endsWith(".zip"))
+            // TODO: notify supplier about non-zip files
+            .forEach(cloudBlob -> {
+                File tempFile = null;
+                try {
+                    // let's download every zip file using temporary file
+                    tempFile = File.createTempFile(cloudBlob.getContainer().getName(), "");
```

Comment:
We could use streams instead of storing it to disk and fetching it again back before sending it to doc store. IO intensive operation. Do we really need to do this?